src/services/expense.service.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+const { Expense } = require('../database');
+
+const getAllExpenses = async(req) => {
+  const { userId, categories, from, to } = req.query;
+  const expensesX = await Expense.findAll();
+
+  let filteredExpenses = [...expensesX];
+
+  if (categories && userId) {
+    filteredExpenses = expensesX.filter(item => {
```

Comment:
I would strongly recommend to let the database filter the collection (it's gonna be waaay more efficient). At this point, the entire filter logic is on the server.

We can achieve it like this:
```javascript
const whereClause = {};

if(userId) {
  whereClause.userId = userId
}

// and so on...
```

And then, at the very end
```javascript
Expense.findAll({ where: whereClause });
```